app/src/screens/locations/LocationDetails.tsx

```diff
@@ -0,0 +1,158 @@
+import { Ionicons } from '@expo/vector-icons'
+import { NativeStackScreenProps } from '@react-navigation/native-stack'
+import React, { useEffect } from 'react'
+import { Layout, Section, SectionContent, Text, TopNav, themeColor } from 'react-native-rapi-ui'
+import { FlatList, Image, StyleSheet, View } from 'react-native'
+import { MainStackParamList } from '../../types/navigation'
+import { StarRatingDisplay } from 'react-native-star-rating-widget'
+import CommentCard from './CommentCard'
+import { CommentsService, IComment, IProfile } from '../../services/comments/commentsService'
+import Loading from '../utils/Loading'
+import { useQuery } from '@tanstack/react-query'
+
+interface ILocation {
+  created_at: string,
+  cep: string,
+  name: string,
+  state: string,
+  number: string,
+  complement: string,
+  city: string,
+  district: string,
+  latitude: string,
+  longitude: string,
+  id: string,
+  company_id: string,
+  image_url: string
+  openning_hour: string
+  closing_hour: string
+}
+
+interface ILocationProps extends NativeStackScreenProps<MainStackParamList, 'LocationDetails'> {
+}
+
+export default function LocationDetails(props: ILocationProps) {
+  const { navigation, route } = props
+  const location = (route?.params || {}) as ILocation
+  function useComments(id: string) {
+    return useQuery ({
+      queryKey: ['comment', id],
+      queryFn: () => CommentsService.getAllCommentsFromLocation(id)
+    })
+  }
+  const { error, data, isLoading } = useComments(location.id)
+  useEffect(() => {
+    if (error) {
+      console.log(error)
+    }
+  })
+
+  const getLocationAverageRating = (comments: IComment[]) => {
+    return (comments.reduce((avaliationSum, item) => avaliationSum + (item?.avaliation || 0), 0) / comments.length)
+  }
+
+  const formatDataToComments = (item: IComment & { profiles: IProfile }) => {
+    return ({
+      profile: {
+        name: item.profiles.name,
+        img_url: item.profiles?.img_url || ''
+      },
+      comment: item.comment,
+      avaliation: item.avaliation
+    })
+  }
+  const isOpen = (location: ILocation) => {
+    const now = new Date()
+
+    const day = now.getDate()
+    const month = now.getUTCMonth() + 1
+    const year = now.getFullYear()
+    const openningDate = new Date(`${year}-${month}-${day} ` + location.openning_hour)
+    const closingDate = new Date(`${year}-${month}-${day} ` + location.closing_hour)
```

Comment:
can you add `luxon` to handle with dates in the project?

https://www.npmjs.com/package/luxon